### PR TITLE
[IRON-23904] upgrade np to v8.0.4 to fix npm release bug

### PR DESCRIPTION
**Pull request by agillaspie**
# IRON-23904

## Description
When trying to run `npm release` on Node.js 18.x, we would run into the following error:
```
npm ERR! ls-collaborators is not a valid access command
```

This command (`npm access ls-collaborators`) was deprecated in `npm 7.x`, but the `np` package was still using it. We needed to upgrade to `np@8.0.4` where this issue is resolved.

We also had to make some minor changes to the `package.json` file as a result of upgrading `np`.

## Manual Testing
Verified that we can successfully use `np@8.0.4` to create + publish a release on **Node.js 18.18.2**.

Use the following command to test the release without actually publishing files:
```
node_modules/.bin/np --preview --any-branch
```

[IRON-23904]: https://ironcladapp.atlassian.net/browse/IRON-23904?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,9 @@
     "lib",
     "umd"
   ],
+  "engines": {
```
Comment (mewelling) [inline package.json:30]: We this is a client side react-lib and `np` is a devDependency, let's remove this engines block. Also means we could get by calling this a patch update vs a major version

**Review by timothymorse — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (agillaspie) [reply package.json:30]: np requires the engines block unfortunately. it's not enforced unless we add `engine-strict: true` as well.

how about we just drop it down to `node: >=14.0.0` for good measure?

Comment (mewelling) [reply package.json:30]: makes sense!

**Review by mewelling — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
     "test:coverage": "jest --coverage",
     "test:coverage-watch": "jest --coverage --watch",
     "lint": "./node_modules/.bin/eslint src/ tests/ demo/src/ --fix",
-    "release": "np",
+    "release": "./node_modules/.bin/np",
```
Comment (brdly-smith) [inline package.json:20]: just curious why this is needed, I thought anything in scripts automatically pulls in node_module .bin?

**Review by brdly-smith — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (mewelling) [reply package.json:20]: i thought so, too, but also see we've done it historically with `lint`, so ¯\_(ツ)_/¯

Comment (agillaspie) [reply package.json:20]: the `np` package is typically installed globally. so this is to make sure that we're using the correct version that we've pinned in this package.json

**Pull request merged**